docs/incubating/custom-evaluator.md

```diff
@@ -0,0 +1,70 @@
+# Custom violations evaluator (incubating)
+
+> Since: [`0.5.1`](https://github.com/novoda/gradle-static-analysis-plugin/releases/tag/v0.5.1)
+>
+> :warning: This is an **experimental**, incubating feature that may be subject to breaking API changes at any time!
+
+The plugin uses a [`ViolationsEvaluator`][violationsevaluatorcode] to determine what to do with the results collected from all the active
+tools (if any). The built-in behaviour is provided by the [`DefaultViolationsEvaluator`][defaultviolationsevaluatorcode], which you can
+read more about [below](#the-defaultviolationsevaluator). The plugin's violations evaluation behaviour is not fixed, and it can be
+customised by providing an implementation of the [`ViolationsEvaluator`][violationsevaluatorcode] interface.
+
+## Table of contents
+ * [The `DefaultViolationsEvaluator`](#the-defaultviolationsevaluator)
+ * [Creating a custom violations evaluator](#creating-a-custom-violations-evaluator)
+
+---
+
+## The `DefaultViolationsEvaluator`
+The plugin has a default mechanism to decide whether to consider a build as passing or failed. The mechanism is manifesting itself
+as the `penalties` closure:
+
+```gradle
+staticAnalysis {
+    penalties {
+        maxErrors = 0
+        maxWarnings = 10
+    }
+    //...
+}
+```
+
+This closure instructs the plugin to use a [`DefaultViolationsEvaluator`][defaultviolationsevaluatorcode] that will count the number of
```

Comment:
It's worth noting that the default evaluator ships with a default configuration, that is you don;t need to specify to define `penalty{}` and you will roll with `DefaultViolationsEvaluator` using the default thresholds (see [`PenaltyExtension.groovy#L5-L6`](https://github.com/novoda/gradle-static-analysis-plugin/blob/master/plugin/src/main/groovy/com/novoda/staticanalysis/PenaltyExtension.groovy#L5-L6))

Reply:
f41c03f7144286431a651708a3f949390e4304ab